drivers/sensor/mcp9808/mcp9808.c

```diff
@@ -78,10 +111,16 @@ int mcp9808_init(const struct device *dev)
 
 	data->i2c_master = device_get_binding(cfg->i2c_bus);
 	if (!data->i2c_master) {
-		LOG_DBG("mcp9808: i2c master not found: %s", cfg->i2c_bus);
+		LOG_ERR("mcp9808: i2c master not found: %s", cfg->i2c_bus);
 		return -EINVAL;
 	}
 
+	rc = mcp9808_set_temperature_resolution(dev, cfg->resolution);
+	if (!rc) {
```

Comment:
This eval will be true for rc == 0, which is not error. Have you tested these changes?

Reply:
This was changed at request: https://github.com/zephyrproject-rtos/zephyr/pull/31758#discussion_r568488344

Please confirm if this is to be set as an error or not. If not, I'll put it back to log debug.

Reply:
the problem is not `LOG_ERR`, this branch will be executed if rc == 0, i.e. operation succeeds.

Reply:
After testing with logging enabled, I fully see what you mean. This will be updated.